core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -3354,8 +3354,6 @@ public synchronized void doConfigSubmit( StaplerRequest req, StaplerResponse rsp
 
             systemMessage = Util.nullify(req.getParameter("system_message"));
 
-            setJDKs(req.bindJSONToList(JDK.class, json.get("jdks")));
```

Comment:
This call is no more needed as the JDKs list is set in a call to its descriptor `configure` in `GlobalToolConfiguration.doConfigure`. Moreover it was clearing out the JDKs list on each general configuration submit (because there is no such `jdks` attribute in the JSON now).